addons/mail/models/mail_thread.py

```diff
@@ -2321,6 +2472,62 @@ def _message_create(self, values_list):
             clean_context(self.env.context)
         ).create(create_values_list)
 
+    def _get_invalid_field_names(self):
+        return [
+            'author_avatar',
+            'canned_response_ids',
+            'child_ids',
+            'mail_ids',
+            'notification_ids',
+            'notified_partner_ids',
+            'reaction_ids',
+            'starred_partner_ids',
+        ]
+
+    def _get_source_from_ref(self, source_ref):
+        """ From a source_reference, return either a mail template, either
+        an ir ui view. """
+        template_id, view_ref = False, False
+        if isinstance(source_ref, models.BaseModel):
+            if source_ref._name == 'mail.template':
+                template_id, view_ref = source_ref.id, False
+            elif source_ref._name == 'ir.ui.view':
+                template_id, view_ref = False, source_ref.id
+            else:
+                raise ValueError(
+                    _('Invalid template or view source record %(svalue)s, is %(model)s instead',
+                       svalue=source_ref,
+                       model=source_ref._name,
+                    ))
+        elif isinstance(source_ref, str):
+            res_model, res_id = self.env['ir.model.data']._xmlid_to_res_model_res_id(source_ref, raise_if_not_found=True)
+            if res_model == 'mail.template':
+                template_id, view_ref = res_id, False
+            elif res_model == 'ir.ui.view':
+                template_id, view_ref = False, res_id
+            else:
+                raise ValueError(
+                    _('Invalid template or view source reference %(svalue)s, is %(model)s instead',
+                       svalue=source_ref,
+                       model=res_model,
+                    ))
+        else:
+            raise ValueError(
+                _('Invalid template or view source %(svalue)s (type %(stype)s), should be a record or an XMLID',
+                  svalue=source_ref,
+                  stype=type(source_ref),
+                ))
+        return template_id, view_ref
+
+    def _raise_for_invalid_fields(self, values_list, invalid_field_names):
+        for values in values_list:
+            conflicting_field_names = set(invalid_field_names) & set(values.keys())
+            if conflicting_field_names:
+                raise ValueError(
```

Comment:
To consider if you do want to be exhaustive in the error message?
```suggestion
        conflicting_field_names = set()
        invalid_field_names = set(invalid_field_names)
        for values in values_list:
            conflicting_field_names |= invalid_field_names & values.keys()
        if conflicting_field_names:
            raise ValueError(
```